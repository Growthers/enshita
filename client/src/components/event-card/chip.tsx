import { memo } from "react";
import styles from "~/styles/components/event-card/chip.module.scss"
import type { ChipProperties } from "./type/model";

// eslint-disable-next-line react/display-name,react/prop-types
const Chip = memo<ChipProperties>(({ status, ...rest}) => {
  let color;
  let content;
  switch(status) {
    case "preparing":
      color = styles["enshita-event-card-status-gray"]
      content = "参加登録受付前"
      break
    case "open":
      color = styles["enshita-event-card-status-green"]
      content = "参加登録受付中"
      break
    case "close":
      color = styles["enshita-event-card-status-gray"]
      content = "参加登録締切済み"
      break
    case "suddenOpen":
      color = styles["enshita-event-card-status-green"]
      content = "飛び入り参加登録受付中"
      break
    case "suddenClose":
      color = styles["enshita-event-card-status-gray"]
      content = "飛び入り参加登録受付終了"
      break
    case "finish":
      color = styles["enshita-event-card-status-gray"]
      content = "イベント終了"
      break
    // no default
  }
  return (
    <div className={`${styles["enshita-event-card-chip"]} ${color}`} {...rest}>
      <p className={styles["enshita-event-card-chip-content"]}>{content}</p>
    </div>
  )
})

export { Chip }
