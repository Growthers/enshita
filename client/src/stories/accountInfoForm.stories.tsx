// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
