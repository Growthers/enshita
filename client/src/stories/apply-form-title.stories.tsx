import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
