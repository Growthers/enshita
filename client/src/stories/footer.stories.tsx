// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/layout/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
