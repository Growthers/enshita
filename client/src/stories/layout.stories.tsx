import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Layout from "~/components/layout";

type T = typeof Layout;
type Story = ComponentStory<T>;

export default {
  title: "Layout",

  component: Layout,
} as ComponentMeta<T>;

export const Default: Story = () => (
  <Layout>
    <div>hogehoge</div>
  </Layout>
);
