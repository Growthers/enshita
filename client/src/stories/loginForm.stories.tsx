// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import LoginForm from "~/components/loginForm";

type T = typeof LoginForm;
type Story = ComponentStoryObj<T>;

export default {
  title: "LoginForm",
  component: LoginForm,
} as ComponentMeta<T>;

export const Default: Story = {};
