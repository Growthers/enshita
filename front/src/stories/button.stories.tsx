import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Button from "~/components/button";

type T = typeof Button;
type Story = ComponentStoryObj<T>;

export default {
  title: "Button",
  args: { children: <div>hogehoge</div> },
  component: Button,
} as ComponentMeta<T>;

export const Default: Story = {};
