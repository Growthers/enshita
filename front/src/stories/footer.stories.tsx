import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
