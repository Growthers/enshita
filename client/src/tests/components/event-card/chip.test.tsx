import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/event-card/chip.stories";

const { Default, Open, Close, SuddenOpen, SuddenClose, Finish } =
  composeStories(stories);

describe("(components) event card chip", () => {
  test("(normal) take snapshot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
  test("(open) take snapshot", () => {
    const { container } = render(<Open />);
    expect(container).toMatchSnapshot();
  });
  test("(close) take snapshot", () => {
    const { container } = render(<Close />);
    expect(container).toMatchSnapshot();
  });
  test("(sudden open) take snapshot", () => {
    const { container } = render(<SuddenOpen />);
    expect(container).toMatchSnapshot();
  });
  test("(sudden close) take snapshot", () => {
    const { container } = render(<SuddenClose />);
    expect(container).toMatchSnapshot();
  });
  test("(finish) take snapshot", () => {
    const { container } = render(<Finish />);
    expect(container).toMatchSnapshot();
  });
});
