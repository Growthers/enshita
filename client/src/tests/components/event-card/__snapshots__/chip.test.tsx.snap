// Vitest Snapshot v1

exports[`(components) event card chip > (close) take snapshot 1`] = `
<div>
  <div
    class="_enshita-event-card-chip_f96697 _enshita-event-card-status-gray_f96697"
  >
    <p
      class="_enshita-event-card-chip-content_f96697"
    >
      参加登録締切済み
    </p>
  </div>
</div>
`;

exports[`(components) event card chip > (finish) take snapshot 1`] = `
<div>
  <div
    class="_enshita-event-card-chip_f96697 _enshita-event-card-status-gray_f96697"
  >
    <p
      class="_enshita-event-card-chip-content_f96697"
    >
      イベント終了
    </p>
  </div>
</div>
`;

exports[`(components) event card chip > (normal) take snapshot 1`] = `
<div>
  <div
    class="_enshita-event-card-chip_f96697 _enshita-event-card-status-gray_f96697"
  >
    <p
      class="_enshita-event-card-chip-content_f96697"
    >
      参加登録受付前
    </p>
  </div>
</div>
`;

exports[`(components) event card chip > (open) take snapshot 1`] = `
<div>
  <div
    class="_enshita-event-card-chip_f96697 _enshita-event-card-status-green_f96697"
  >
    <p
      class="_enshita-event-card-chip-content_f96697"
    >
      参加登録受付中
    </p>
  </div>
</div>
`;

exports[`(components) event card chip > (sudden close) take snapshot 1`] = `
<div>
  <div
    class="_enshita-event-card-chip_f96697 _enshita-event-card-status-gray_f96697"
  >
    <p
      class="_enshita-event-card-chip-content_f96697"
    >
      飛び入り参加登録受付終了
    </p>
  </div>
</div>
`;

exports[`(components) event card chip > (sudden open) take snapshot 1`] = `
<div>
  <div
    class="_enshita-event-card-chip_f96697 _enshita-event-card-status-green_f96697"
  >
    <p
      class="_enshita-event-card-chip-content_f96697"
    >
      飛び入り参加登録受付中
    </p>
  </div>
</div>
`;
